cmake/detect-intrinsics.cmake

```diff
@@ -540,13 +503,12 @@ macro(check_xsave_intrinsics)
     set(CMAKE_REQUIRED_FLAGS "${XSAVEFLAG} ${NATIVEFLAG}")
     check_c_source_compiles(
         "#ifdef _WIN32
-         #  include <intrin.h>
-         #else
-         #  include <x86gprintrin.h>
-         #endif
-         int main(void) {
-             return _xgetbv(0);
-         }"
+        #  include <intrin.h>
+        #else
+        #  include <x86gprintrin.h>
+        #endif
+        unsigned int f(unsigned int a) { return _xgetbv(a); }
```

Comment:
`_xgetbv` returns `unsigned __int64`
https://www.intel.com/content/www/us/en/docs/cpp-compiler/developer-guide-reference/2021-8/xgetbv.html
```
 warning C4244: 'return': conversion from 'unsigned __int64' to 'unsigned int', possible loss of data
```

Reply:
It's no change from before where it was returned from main. Where is `__int64` defined? Doesn't look standard to me.

Reply:
The warning is from MSVC. I think if you just cast it, it should be fine.